spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/session/ReactiveSessionsEndpointWebIntegrationTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.session;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.boot.actuate.endpoint.web.test.WebEndpointTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.session.MapSession;
+import org.springframework.session.ReactiveSessionRepository;
+import org.springframework.session.Session;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Integration tests for {@link ReactiveSessionsEndpoint} exposed by WebFlux.
+ *
+ * @author Vedran Pavic
+ */
+class ReactiveSessionsEndpointWebIntegrationTests {
+
+	private static final Session session = new MapSession();
+
+	@SuppressWarnings("unchecked")
+	private static final ReactiveSessionRepository<Session> sessionRepository = mock(ReactiveSessionRepository.class);
+
+	@WebEndpointTest
```

Comment:
@wilkinsona is there a way to have `@WebEndpointTest` run only on some endpoint implementations, like WebFlux in this case? If not, what is the preferred way of writing actuator endpoint tests that target only a specific implementation?

Reply:
There isn't.

One option would be to copy-paste `WebEndpointTestInvocationContextProvider.WebFluxEndpointConfiguration` into these tests and modify it as needed. However, I think it might be better if `WebEndpointTest` had an attribute that allowed the tests to be run against particular web frameworks rather than always running against all three of them.

Reply:
Thanks for the early feedback - I've opened https://github.com/spring-projects/spring-boot/issues/32054 to consider enhancing `@WebEndpointTest` as suggested.

Reply:
Thanks. Could you please try the changes in https://github.com/wilkinsona/spring-boot/tree/gh-32054 and see if they meet your needs here?

Reply:
I did, and outside of Checkstyle bugging me about redundant `public` modifier [here](https://github.com/wilkinsona/spring-boot/blob/7d64e9496a425aed3af5c9e543fcf212b5d8deb9/spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/endpoint/web/test/WebEndpointTest.java#L50), everything was great. I've updated the PR and pushed those changes.

I hope to continue working on the PR tomorrow and expect to wrap up the changes by the end of the week.